TwoNote/app/src/main/java/com/microsoft/device/display/samples/twonote/FileHandler.kt

```diff
@@ -0,0 +1,121 @@
+package com.microsoft.device.display.samples.twonote
+
+import android.content.Context
+import android.util.Log
+import com.microsoft.device.display.samples.twonote.model.DataProvider
+import com.microsoft.device.display.samples.twonote.model.DirEntry
+import com.microsoft.device.display.samples.twonote.model.INode
+import com.microsoft.device.display.samples.twonote.model.Note
+import java.io.File
+import java.io.FileInputStream
+import java.io.FileOutputStream
+import java.io.ObjectInputStream
+import java.io.ObjectOutputStream
+import java.lang.Exception
+
+class FileHandler {
+    // loads inode information from the current directory into the DataProvider
+    fun loadDirectory(context: Context, subDir: String) {
+        if (DataProvider.inodes.isEmpty()) {
+            readDirEntry(context, subDir)?.let { notes ->
+                for (inode in notes.inodes) {
+                    DataProvider.addINode(inode)
+                }
+            }
+        }
+    }
+
+    // add a new inode
+    fun addInode(subDir: String): Int {
+        val inode = INode()
+        if (DataProvider.inodes.isNotEmpty()) {
+            inode.id = DataProvider.inodes[DataProvider.inodes.lastIndex].id + 1
+            inode.title = "Note " + inode.id
+            DataProvider.addINode(inode)
+            return DataProvider.inodes.lastIndex
+        } else {
+            DataProvider.addINode(inode)
+            return DataProvider.inodes.lastIndex
+        }
+    }
+
+    // reads a file and parses note data
+    fun loadNote(context: Context, subDir: String, noteName: String): Note? {
+        val path: String? = context.getExternalFilesDir(null)?.absolutePath
+        val file = File(path + subDir + noteName)
+        var fileStream: FileInputStream? = null
+        var objectStream: ObjectInputStream? = null
+
+        try {
+            fileStream = FileInputStream(file)
+            objectStream = ObjectInputStream(fileStream)
+            val note = objectStream.readObject()
+            if (note is Note) {
+                return note
+            } else {
+                Log.e(this.javaClass.toString(), "Error: loaded file is not of type Note")
```

Comment:
this can be done at a later date but we should put strings in the strings.xml file!